// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


enum IamRoleStageEnum {
  IamRoleStageEnumNO_VALUE_DO_NOT_USE = 0;
  IamRoleStageEnumALPHA = 1;
  IamRoleStageEnumBETA = 2;
  IamRoleStageEnumGA = 3;
  IamRoleStageEnumDEPRECATED = 4;
  IamRoleStageEnumDISABLED = 5;
  IamRoleStageEnumEAP = 6;
}

message IamRole {
  string name = 1;
  string title = 2;
  string description = 3;
  IamRoleLocalizedValues localized_values = 4;
  string lifecycle_phase = 5;
  string group_name = 6;
  string group_title = 7;
  repeated string included_permissions = 8;
  IamRoleStageEnum stage = 9;
  string etag = 10;
  bool deleted = 11;
  repeated string included_roles = 12;
  string parent = 13;
}

message IamRoleLocalizedValues {
  string localized_title = 1;
  string localized_description = 2;
}

message ApplyIamRoleRequest {
  IamRole resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteIamRoleRequest {
  string service_account_file = 1;
  IamRole resource = 2;
}

message ListIamRoleRequest {
  string service_account_file = 1;
  string Parent = 2;
}

message ListIamRoleResponse {
  repeated IamRole items = 1;
}

service IamRoleService {
  rpc ApplyIamRole(ApplyIamRoleRequest) returns (IamRole);
  rpc DeleteIamRole(DeleteIamRoleRequest) returns (google.protobuf.Empty);
  rpc ListIamRole(ListIamRoleRequest) returns (ListIamRoleResponse);
}
